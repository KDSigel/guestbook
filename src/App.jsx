import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import Header from "./components/Header/Header";
import PrivateRoute from './components/PrivateRoute/PrivateRoute';
import Auth from "./views/Auth/Auth";
import Home from "./views/Home/Home";

export default function App() {
  return (
  <Router>
    <Header />
    <Switch>
      <Route path='/login'>
        <Auth />
      </Route>
      <PrivateRoute path='/'>
        <Home />
      </PrivateRoute>
    </Switch>
  </Router>
  )
}
