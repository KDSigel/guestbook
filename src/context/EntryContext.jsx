import { useContext, createContext, useState } from 'react'

const EntryContext = createContext()

// make a provider
const EntryProvider = ({children}) => {
    const [entries, setEntries] = useState([])
    return <EntryContext.Provider value={{entries, setEntries}}>{children}</EntryContext.Provider>
}

// make a customer hook
const useEntry = () => {
    const context = useContext(EntryContext)
    if (context === undefined) {
        throw new Error('call useEntry hook within a EntryContext Provider');
    }
    return context
}

// export the custom hook and the provider?

export {EntryProvider, useEntry}