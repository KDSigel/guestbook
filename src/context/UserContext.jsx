import { useContext, createContext, useState } from 'react';
import { useHistory } from "react-router-dom";

const UserContext = createContext();

// make a provider
const UserProvider = ({ children }) => {
  const [user, setUser] = useState('');
  const history = useHistory();

  const login = (name, password) => {
    const loginSuccessful =
      name === process.env.AUTH_USERNAME &&
      password === process.env.AUTH_PASSWORD;
    if (loginSuccessful) setUser(name);
    return loginSuccessful;
  };

  const logout = () => {
    setUser('');
  }

  return (
    <UserContext.Provider value={{ user, setUser, login, logout }}>
      {children}
    </UserContext.Provider>
  );
};

// make a customer hook
const useUser = () => {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error('call useUser hook within a UserContext Provider');
  }
  return context;
};

// export the custom hook and the provider?

export { UserProvider, useUser };
