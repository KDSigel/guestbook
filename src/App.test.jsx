import React from 'react'
import { render } from '@testing-library/react'
import App from './App.jsx'
import { EntryProvider } from './context/EntryContext.jsx'
import { UserProvider } from './context/UserContext.jsx'

it('should check App', () => {
const { container } = render(
    <EntryProvider>
    <UserProvider>
        <App />
    </UserProvider>
    </EntryProvider>
)
expect(container).toMatchSnapshot()
}
)