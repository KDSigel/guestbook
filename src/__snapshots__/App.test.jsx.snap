// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should check App 1`] = `
<div>
  <div>
    Log in already
  </div>
  <div>
    <h2>
      Sign the guestbook, now!
    </h2>
    <form>
      <div>
        Guest Name
        <input
          placeholder="Guest Name..."
          value=""
        />
      </div>
      <div>
        Guest Entry
        <textarea
          placeholder="write something meaningful"
        />
      </div>
      <button>
        Sign
      </button>
      
    </form>
    <ul />
  </div>
</div>
`;
